src/request.c

```diff
@@ -1239,12 +1239,13 @@ static S3Status setup_curl(Request *request,
     // less than 1K per second for more than 15 seconds.
     // xxx todo - make these configurable
     // xxx todo - allow configurable max send and receive speed
-    curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, 1024);
-    curl_easy_setopt_safe(CURLOPT_LOW_SPEED_TIME, 15);
-
+    // jjames - only set the low speed limit and time if the timeout is 0
 
     if (params->timeoutMs > 0) {
         curl_easy_setopt_safe(CURLOPT_TIMEOUT_MS, params->timeoutMs);
+    } else {
+        curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, 256);
```

Comment:
why the change from 1024 above to 256 here?

otherwise, i think your comment is justification enough for moving it into the else block.